### PR TITLE
rewriting laundry

### PR DESCRIPTION
**Pull request by ankit204**
I'm hitting a different url for data which doesn't require cookies.

### CODE REVIEW

File: PittAPI/laundry.py [hunk 1]
```diff
@@ -1,30 +1,17 @@
...
-
-This program is distributed in the hope that it will be useful,
-but WITHOUT ANY WARRANTY; without even the implied warranty of
-MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-GNU General Public License for more details.
-
-You should have received a copy of the GNU General Public License along
-with this program; if not, write to the Free Software Foundation, Inc.,
-51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+    File: laundry.py
```
Comment (RitwikGupta) [inline PittAPI/laundry.py]: Don't modify the license string

File: PittAPI/laundry.py [hunk 2]
```diff
@@ -36,10 +23,10 @@
 }
 
 
-def get_status_simple(building_name):
+def get_overall_status(building_name):
```
Comment (RitwikGupta) [inline PittAPI/laundry.py]: No need to change the method identifier.

File: PittAPI/laundry.py [hunk 3]
```diff
@@ -49,92 +36,85 @@ def get_status_simple(building_name):
...
-
-    rg = re.compile(''.join(re1), re.IGNORECASE | re.DOTALL)
-    search = rg.findall(str(soup))
-
-    di = {
+    url ='http://m.laundryview.com/submitFunctions.php?monitor=true&lr={}'.\
+            format(location_lookup[building_name])
+    response = requests.get(url)
+    laundry_soup = BeautifulSoup(response.text, 'lxml')
+    reFormat = re.compile(r'^([0-9]+) of ([0-9]+) available$')
```
Comment (RitwikGupta) [inline PittAPI/laundry.py]: Does not follow PEP8 naming conventions

File: PittAPI/laundry.py [hunk 4]
```diff
@@ -49,92 +36,85 @@ def get_status_simple(building_name):
...
-        if machine_status is u'In use':
-            time_left = int(machine[1])
-        else:
-            time_left = -1 if machine[6] is '' else machine[6]
-        di.append({
+        machines.append({
             'machine_name': machine_name,
+            'machine_type': machine_type,
             'machine_status': machine_status,
-            'time_left': time_left
```
Comment (RitwikGupta) [inline PittAPI/laundry.py]: `time_left` is critical. Do not take this away.

File: PittAPI/laundry.py [hunk 5]
```diff
@@ -49,92 +36,85 @@ def get_status_simple(building_name):
...
+        -> MCCORMICK
+        -> SUTH_EAST
+        -> SUTH_WEST
+    :param machine: (String) specific machine to query
+                    eg: machine='BA02'
+    """
+    building_name = building_name.upper()
+    machines = []
+
+    url = 'http://m.laundryview.com/submitFunctions.php?monitor=true&lr={}'.\
```
Comment (RitwikGupta) [inline PittAPI/laundry.py]: The backslash should be before the period, and the period should go on the next line with format, so it becomes `.format`

File: PittAPI/laundry.py [hunk 6]
```diff
@@ -49,92 +36,85 @@ def get_status_simple(building_name):
...
-    soup = BeautifulSoup(page.text, 'lxml')
-
-    # What was I doing...
-    re1 = ['(\\d+)', '(\\s+)', '(of)', '(\\s+)', '(\\d+)', '(\\s+)', '((?:[a-z][a-z]+))']
-
-    rg = re.compile(''.join(re1), re.IGNORECASE | re.DOTALL)
-    search = rg.findall(str(soup))
-
-    di = {
+    url ='http://m.laundryview.com/submitFunctions.php?monitor=true&lr={}'.\
```
Comment (RitwikGupta) [inline PittAPI/laundry.py]: It should be like
```
{}'\
    .format
```
Slash before period, period with method

File: PittAPI/laundry.py (hunk 1 above)
Comment (ankit204) [reply PittAPI/laundry.py]: Why's that?

Comment (timparenti) [reply PittAPI/laundry.py]: The GPL explicitly states "You may copy and distribute verbatim copies of the Program's source code as you receive it, in any medium, provided that you conspicuously and appropriately publish on each copy an appropriate copyright notice and disclaimer of warranty; keep intact all the notices that refer to this License and to the absence of any warranty; and give any other recipients of the Program a copy of this License along with the Program."

In short, the license requires that, if you use/modify this code, you not change the license.

File: PittAPI/laundry.py (hunk 4 above)
Comment (ankit204) [reply PittAPI/laundry.py]: Well we have a `machine_status` that indicates the number of minutes left when it is being used.  It is `Avail` if idle (and online). Not sure about the significance of a separate field.

**Comment by Vinny55** [comment]
Hey I am no longer at Pitt of The CSC, anyway to remove me from the group
so I dont see future emails like these?

Thanks,
Vincent Quezada

File: PittAPI/laundry.py (hunk 4 above)
Comment (dzheng256) [reply PittAPI/laundry.py]: `time_left` is the number of minutes until it's finished

Comment (RitwikGupta) [reply PittAPI/laundry.py]: Having a string showing the whole thing is not as useful as having an explicit field with an integer, which is why that field is there. Also, this would introduce breaking API changes.

**Comment by RitwikGupta** [comment]
@ankit204 any update on this PR?
